packages/protocol/src/utils/test-util.spec.ts

```diff
@@ -0,0 +1,45 @@
+/********************************************************************************
```

Comment:
I think it is a bit unexpected to have a `test-util.spec.ts` without a `test-util.ts`. Could you explain what this file tests? If it is only the `MaybePromise` I believe we should add it to the `type-util.spec.ts`, what do you think?

Reply:
This modules provides utility methods that are only used in the test context (`expectToThrowAsync`). It imports from `chai` which is a test dependency => it should not be part of the production build. That`s is the reason why I used the `spec` suffix. 
While I agree that the `spec` suffix is a bit unexpected I don't think moving it into the `type-util.spec.ts` is a good idea.
I would rather drop the spec suffix and exclude the file explicitly from the tsconfig.

Reply:
Sounds good and works for me, thank you for the fast response!